resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesV1Suite.scala

```diff
@@ -171,6 +171,22 @@ private[spark] class KubernetesV1Suite(testBackend: IntegrationTestBackend)
       "Unexpected value for annotation2")
   }
 
+  test("Run with driver pod name") {
```

Comment:
I wish we did not add more e2es and instead used unit tests better. This one is fine for now, but it's something to consider.